packages/experience-builder-sdk/src/blocks/CompositionBlock.tsx

```diff
@@ -66,7 +96,7 @@ export const CompositionBlock = ({
         }
         case 'UnboundValue': {
           const uuid = variable.key;
-          acc[variableName] = unboundValues[uuid]?.value;
+          acc[variableName] = (entityStore?.unboundValues || unboundValues)[uuid]?.value;
```

Comment:
But Daniel, you wrote the code that stores the unbound values in the preview/delivery mode entity store [here](https://github.com/contentful/experience-builder/blob/feat/render-design-component-within-an-experience/packages/experience-builder-sdk/src/core/EntityStore.ts#L46) 😂.

The unbound values props passed into this component comes from the store [here](https://github.com/contentful/experience-builder/blob/79031a2ddf46adefad40055852f7e21d5e58fd81/packages/experience-builder-sdk/src/blocks/PreviewDeliveryRoot.tsx#L51)

I also don't see any problems with storing them in the entity store. I'm using it directly here because I added an update unbound values method [here](https://github.com/contentful/experience-builder/pull/151/files/cc74cd52fc4452978b94faa6f64e6264f3249381) that I use to update the unbound values object when deserializing design component nodes [here](https://github.com/contentful/experience-builder/pull/151/files/cc74cd52fc4452978b94faa6f64e6264f3249381#diff-966851c3ad5b7a4330fcc1f676a4c2a9583fab1a6d57266c0ddd50b9a6acb3bbR38)

Reply:
Also `CompositionBlock` is not used in editor mode, it's used only for preview/delivery mode. We use the `VisualEditorBlock` in editor mode, so the stores are correctly different and supply different things.